redis/conn.go

```diff
@@ -137,6 +148,21 @@ func Dial(network, address string, options ...DialOption) (Conn, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if do.dialTLS {
+		tlsConfig := do.tlsConfig
+		if tlsConfig == nil {
+			// https://golang.org/pkg/crypto/tls/#Client
+			// At this point we don't know the ServerName.
+			tlsConfig = &tls.Config{InsecureSkipVerify: true}
```

Comment:
There are some other alternatives here:

(1)
```go
if do.tlsConfig == nil {
  return nil, errors.New("a valid *tls.Config is required")
```

(2)
```go
if do.tlsConfig == nil {
  if do.serverName == "" {
    return nil, errors.New("a valid tlsConfig or serverName is required")
  }
  tlsConfig = &tls.Config{ServerName: do.serverName}
}
```

The later requires adding a `serverName` option to dialOptions.


TBH: I don't like either, but nor do I like InsecureSkipVerify:true (but see previous comments on why I think it's sanest atm).